root/usr/share/fdi/facts/discovery-facts.rb

```diff
@@ -161,3 +162,41 @@ def add_ipmi_facts fact, value, channel = nil
     end
   end
 end
+
+# NetworkManager details (e.g. nmprimary_dhcp4_option_domain_name)
+nmout = Facter::Util::Resolution.exec("nmcli -t con show primary 2>/dev/null")
+nmfacts = {}
+nmout.each_line do |x|
+  elements = x.split(":", 2)
+  name = elements.first.downcase.sub(/\[\d+\]$/,"")
+  if name =~ /dhcp.\.option/
+    dhcp_elems = elements.last.split(/\s*=\s*/, 2)
+    name += '_' + dhcp_elems.first
+    value = dhcp_elems.last
+  else
+    value = elements.last
+  end
+  name = "nmprimary_" + name.tr('.', '_')
+  nmfacts[name] = value
+  Facter.add(name) do
+    setcode do
+      value
+    end
+  end
+end
+
+# Create DHCP FQDN helper facts
+Facter.add("nmprimary_dhcp4_option_fqdn") do
+  setcode do
+    nmfacts["nmprimary_dhcp4_option_host_name"] + '.' + nmfacts["nmprimary_dhcp4_option_domain_name"]
+  end
+end
+
+# Primary interface IP PTR DNS record
+ip = nmfacts["nmprimary_ip4_address"].scan(/\d+\.\d+\.\d+\.\d+/).first.strip rescue '0.0.0.0'
```

Comment:
I think, it would be a lot more clearer to rescue `nil` here and check and only run the code below if the ip is not nil. Setting it to `0.0.0.0` is functional the same but might actually issue a dns query that has to come back with no result.